Python/bin/xmltopsv.py

```diff
@@ -300,8 +299,12 @@ def processObsContext(element):
 #
 # read in allowedElementDict, requiredElementDict  and psvFormatDict
 #
+
+# Important note: it is very complicated and time consuming to change this routine.
+# I will keep it as it is and we can only call it through command line
+
 (allowedElementDict, requiredElementDict, psvFormatDict) = \
-                                         adesutility.getAdesTables()
+                                       adesutility.getAdesTables()
 
 inputTree = adesutility.readXML(sys.argv[1])
```

Comment:
I think you just need to wrap two lines in the main function and leave the rest as global variables. How does this look? Although it would be best to not use global variables and pass in `encodeout` and other variables into the functions where they are needed.
```suggestion

allowedObsDataSet = setFromElementDictList('obsData')
allowedObsBlockSet = setFromElementDictList('obsBlock')
allowedAdesSet = setFromElementDictList('ades')
encodeout = None

def xmltopsv(args):
    global encodeout
    inputTree = adesutility.readXML(args[1])
    encodedout = io.open(args[2], 'w', encoding=psvencoding)
    processAdesElement(inputTree.getroot())

if __name__ == "__main__":
    xmltopsv(sys.argv)

```

Reply:
@federicaspoto I've pushed a commit that resolves this using the above suggestion. test_xmltopsv passes, but can you take a look as well?

Reply:
@stevenstetzler thanks! It looks good to me.